lib/src/util/tested_expressions.dart

```diff
@@ -88,9 +88,10 @@ class TestedExpressions {
       return _contradictions;
     }
 
-    final binaryExpression = testingExpression is BinaryExpression
-        ? testingExpression as BinaryExpression
-        : null;
+    var testingExpression = this.testingExpression;
+
+    final binaryExpression =
+        testingExpression is BinaryExpression ? testingExpression : null;
```

Comment:
Consider writing this as
```
var facts;
if (testingExpression is BinaryExpression) {
  facts = [testingExpression.leftOperand, testingExpression.rightOperand];
} else {
  facts = [testingExpression];
}
```
I think it's a little easier to follow the logic.

Reply:
Nice one.  Agreed. 👍